arch/arm/core/cortex_m/tz/CMakeLists.txt

```diff
@@ -18,6 +18,21 @@ if(CONFIG_ARM_FIRMWARE_HAS_SECURE_ENTRY_FUNCS)
   zephyr_ld_options(
     ${LINKERFLAGPREFIX},--cmse-implib
   )
+
+  if(CONFIG_ARM_FIRMWARE_HAS_SECURE_ENTRY_FUNCS)
+    # Add dummy command to "produce" the veneers lib. The file is actually
+    # generated by the linking of ${IMAGE}kernel_elf.
+    add_custom_command(
+      OUTPUT ${CMAKE_BINARY_DIR}/${CONFIG_ARM_ENTRY_VENEERS_LIB_NAME}
+      COMMAND
+      DEPENDS ${IMAGE}kernel_elf)
+    add_custom_target(veneers
+      DEPENDS ${CMAKE_BINARY_DIR}/${CONFIG_ARM_ENTRY_VENEERS_LIB_NAME})
+  endif()
 endif()
 
-zephyr_sources(arm_core_tz.c)
+zephyr_link_libraries_ifdef(CONFIG_ARM_FIRMWARE_USES_SECURE_ENTRY_FUNCS
+  ${CMAKE_BINARY_DIR}/${CONFIG_ARM_ENTRY_VENEERS_LIB_NAME}
+  )
+
+zephyr_sources_ifdef(CONFIG_ARM_SECURE_FIRMWARE arm_core_tz.c)
```

Comment:
I believe, this is now required due to the change in the "parent" CMakeLists.txt, right?

Reply:
Correct.